python/dgl/graphbolt/dataloader.py

```diff
@@ -35,6 +38,52 @@ def _find_and_wrap_parent(
             )
 
 
+class EndMarker(dp.iter.IterDataPipe):
+    """Used to mark the end of a datapipe."""
+
+    def __init__(self, datapipe):
+        self.datapipe = datapipe
+
+    def __iter__(self):
+        for data in self.datapipe:
+            yield data
+
+
+class Bufferer(dp.iter.IterDataPipe):
+    """Buffers items before yielding them."""
+
+    def __init__(self, datapipe, buffer_size=2):
```

Comment:
Add a doc about why buffer_size is default to 2 and how can user decide what's the best for their cases.